src/store/sled.rs

```diff
@@ -805,33 +804,43 @@ impl MessageStore for SledStore {
         }
     }
 
-    fn messages(&self, thread: &Thread, from: Option<u64>) -> Result<Self::MessagesIter, Error> {
+    fn messages(
+        &self,
+        thread: &Thread,
+        range: impl RangeBounds<u64>,
+    ) -> Result<Self::MessagesIter, Error> {
         let tree_thread = self.tree(self.messages_thread_tree_name(thread))?;
         debug!("{} messages in this tree", tree_thread.len());
-        let iter = if let Some(from) = from {
-            tree_thread.range(from.to_be_bytes()..)
-        } else {
-            tree_thread.range::<&[u8], std::ops::RangeFull>(..)
+
+        let iter = match (range.start_bound(), range.end_bound()) {
+            (Bound::Included(start), Bound::Unbounded) => tree_thread.range(start.to_be_bytes()..),
+            (Bound::Included(start), Bound::Excluded(end)) => {
+                tree_thread.range(start.to_be_bytes()..end.to_be_bytes())
+            }
+            (Bound::Included(start), Bound::Included(end)) => {
+                tree_thread.range(start.to_be_bytes()..=end.to_be_bytes())
+            }
+            (Bound::Unbounded, Bound::Included(end)) => tree_thread.range(..=end.to_be_bytes()),
+            (Bound::Unbounded, Bound::Excluded(end)) => tree_thread.range(..end.to_be_bytes()),
+            (Bound::Unbounded, Bound::Unbounded) => tree_thread.range::<[u8; 8], RangeFull>(..),
+            (Bound::Excluded(_), _) => unreachable!("range that excludes the initial value"),
```

Comment:
Why should it not be possible to exclude the first value? E.g. when loading messages later than a message one already has but withou that message, excluding the starting-value would be useful.

Reply:
It's not about what's being useful or not, it's about what one can express with the `a..b` syntax in Rust. AFAIK, you cannot express this.

Reply:
You can use that `RangeBound` is implemented for `(Bound, Bound)` (https://docs.rs/rustc-std-workspace-std/1.0.1/std/ops/trait.RangeBounds.html#foreign-impls). I also do that for Flare.

Reply:
This is nice, thanks (but we still have to do things ourselves here until the method is stabilized).

Reply:
What I have mentioned here is already stable. But as mentioned, nothing really necessary, just a minor comment.